nano/node/node.cpp

```diff
@@ -2668,7 +2668,7 @@ void nano::node::process_confirmed (std::shared_ptr<nano::block> block_a, uint8_
 	auto hash (block_a->hash ());
 	if (ledger.block_exists (block_a->type (), hash))
 	{
-		auto transaction (store.tx_begin_read ());
+		auto transaction (store.tx_begin_write ());
```

Comment:
Would it be possible to to scope everything until add_confirmation_heights in a read tx (as before), and then only do add_confirmation_heights in a write transaction? We want to keep write tx's open as short as possible. confirmed_visitor can take some time if there are many wallets, etc.

Reply:
Good point, done both